src/worldmap.js

```diff
@@ -115,41 +115,99 @@ export default class WorldMap {
     }
   }
 
-  drawPoints() {   
-
+  drawPoints() {
     Object.keys(this.ctrl.data).forEach((layerKey) => {
       let layer = this.ctrl.data[layerKey];
+      let markersGJ = L.geoJSON();
       let markers = L.markerClusterGroup();
 
       //for each layer
       Object.keys(layer).forEach((objectKey) => {
         let lastObjectValues = layer[objectKey][layer[objectKey].length-1];
         lastObjectValues.type = layerKey;
 
-        let newIcon = this.createIcon(lastObjectValues);
-
-        try { 
-          if(newIcon)
-            markers.addLayer(newIcon);            
-        } catch(error) { console.warn(layerKey); console.warn(error); }
+        var geoJsonName = null;
+        var keyArray = Object.keys(lastObjectValues);
+        for (var k = 0; k < keyArray.length; k++) {
+            if(keyArray[k].toLowerCase() === 'geojson'){
+                geoJsonName = keyArray[k];
+                break;
+            }
+        }
+
+        if(geoJsonName && lastObjectValues[geoJsonName]) {
+          let newGJ = this.createGeoJson(lastObjectValues, geoJsonName);
+          newGJ.addTo(markersGJ);
+        }
+        if (lastObjectValues.latitude && lastObjectValues.longitude) {
+          let newIcon = this.createIcon(lastObjectValues, geoJsonName);
+          try {
+            if(newIcon)
+              markers.addLayer(newIcon);
+          } catch(error) { console.warn(layerKey); console.warn(error); }
+        }
       })
 
       this.overlayMaps[layerKey].addLayer(markers);
+      this.overlayMaps[layerKey].addLayer(markersGJ);
+
     });
   }
 
-  createIcon(dataPoint) {
+  createGeoJson(dataPoint, geoJsonName) {
+    var geoColor = this.ctrl.panel.layersColors[dataPoint.type];
+    if (geoColor === 'lightred') {
+        geoColor = '#FF9898';
+    } else if (geoColor === 'darkpurple') {
+        geoColor = '#6813B2';
+    } else if (geoColor === 'black') {
+        geoColor = '#404040';
+    } else if (geoColor === null && this.ctrl.panel.layersIcons[dataPoint.type] !== null) {
+        geoColor = 'red';
+    }
+    var myStyle = {
+      "color": geoColor,
+      "weight": 5,
+      "opacity": 0.65
+    };
+    var retVal;
+    if(typeof dataPoint[geoJsonName] === 'object') {
+        retVal = L.geoJSON(dataPoint[geoJsonName], {
+            style: myStyle
+        });
+    } else {
+        retVal = L.geoJSON(JSON.parse(dataPoint[geoJsonName]), {
+          style: myStyle
+        });
+    }
+    var dataInfoWithoutGeoJson = JSON.parse(JSON.stringify(dataPoint)); //creates clone of json
```

Comment:
why is the clone needed again? is the "delete" if L185 causing a data loss? could we instead adjust the method retrieving the info to not display the geojson?

Reply:
I have re-implemented as you wanted (no cloning, just skip the geojson info in popup).